### ISSUE TITLE
SPIKE: Address potential challenges around homepage soft launch modal 

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#11587]
## Description

Before implementing #11535, we need to understand the mechanics around a few UX/tech risks:
- AUDIT: Competing modals on VAgov – there is already a [modal that appears for users redirected from explore.va.gov](https://www.va.gov/?from=explore.va.gov). We need to see how/where that modal is coded and determine a way to prevent our modal from displaying in that use case.
  - Desired behavior: No other VA.gov modals should appear on page load
  - NEW ISSUE: get rid of exploreVA modal
- Injected header/footer – we need to understand how modals like our soft-launch modal are handled in injected-header/footer scenarios and thus how to ensure it doesn't display anywhere except the actual VAgov homepage
- "don't show me modal twice in on session" logic – if there is any doubt among the team how we would implement a cookie to prevent re-displaying the modal within a single user session, now is the time to explore that. Possible questions relate to:
  - Governance: do we need any other team's buy-in to add a cookie?
  - Do we understand how to define a "user session" in this context? What are our options? (e.g., time-since-interaction, 24 hrs, etc)

## Acceptance Criteria
- [x] Document findings for all three areas in this ticket
- [ ] Review #11535 for compliance with findings

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by ryguyk** [comment]
**(WIP)**

### Findings
The most important finding coming out of this spike is the discovery of **an existing framework for "Announcements"**. The framework allows for the creation of announcements via a config file. [Here is that file at the time of this writing](https://github.com/department-of-veterans-affairs/vets-website/blob/5748c35ff36046476fb8d52c10d2988ced2c3fb5/src/platform/site-wide/announcements/config/index.js).

This framework gives a foundation to discuss how to proceed with each of the three areas/questions of interest outlined in the ticket description:
1. Other modals
    * Audit competing modals on VAgov
        * The config file as it currently exists contains only one entry. There remains the possibility that competing modals would be defined in a separate fashion, but none of those were discovered in this spike. The one entry that is currently there is for the next bullet below...
    * Remove explore.va.gov modal
        * The config file, as it currently exists, defines only the existing modal for the redirect from explore.va.gov. Removing it amounts to removing that entry from the configuration. This is straightforward and has [its own ticket here](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11682).

1. Injected header/footer – we need to understand how modals like our soft-launch modal are handled in injected-header/footer scenarios and thus how to ensure it doesn't display anywhere except the actual VAgov homepage

    * This is a non-issue. The injected header/footer is unrelated for a couple of reasons. First, the code for this announcement framework is not injected. Second, the configuration for a homepage modal should explicitly define the homepage as the desired path. It should not fire on any other pages.

1. "Don't show me modal twice in on session"
    * This is the thing that will require some iteration on the current framework. We have decided to use `sessionStorage` so that a user sees the modal only once per session (tab). The current framework utilizes only `localStorage` via a config property called `showEverytime`. This will need to be expanded. 
    * My recommendation is to change this from a boolean to a string/enum field called, more simply, `show`, with the following options:
        *  `once`
            * Dismissing will hide forever (or at least until a user empties `localStorage`). This is the equivalent of the currently implemented `showEverytime: false`.
        * `every-page-load`
            * Dismissing will hide announcement but it will show again on subsequent page loads. This is the equivalent of the currently implemented `showEverytime: true`. 
        *  `once-per-session`
            * Dismissing will hide until the next time a user visits (in a new tab/session). This is the essence of the new feature.
